### ISSUE TITLE
Cannot Authenticate via LDAP(S)

### ISSUE THREAD

**Issue by bandwiches** [ansible/awx#9906]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY

Cannot authenticate via LDAP(S).

##### ENVIRONMENT
* AWX version: 17.0.1
* AWX install method: docker
* Ansible version:  2.9.17
* Operating System: Ubuntu
* Web Browser: Chrome

##### STEPS TO REPRODUCE
Try to login.

##### EXPECTED RESULTS
To at least

##### ACTUAL RESULTS
Authentication fails and the following log is generated:
`WARNING  django_auth_ldap Caught LDAPError while authenticating user.name: SERVER_DOWN({'result': -1, 'desc': "Can't contact LDAP server", 'errno': 107, 'ctrls': [], 'info': 'Transport endpoint is not connected'},)`

##### ADDITIONAL INFORMATION
ldapsearch works just fine with the same settings.

**Comment by shanemcd** [comment]
Sounds like a config issue. You may have better luck troubleshooting on IRC or the mailing list.

http://webchat.freenode.net/?channels=ansible-awx
https://groups.google.com/forum/#!forum/awx-project

**Comment by bandwiches** [comment]
Riiiiiiiight....thanks I guess

**Comment by bandwiches** [comment]
Is this debatable?  The IRC channel is stale AF.

The error `Transport endpoint is not connected` is of `OSError` origin which points me to the application itself and not the configuration.  In addition, the configuration settings for LDAP are within the AWX_WEB container and not `awx\installer\inventory`.

**Comment by bandwiches** [comment]
Well, great talk.  Glad we took the time to understand what the root issue was here, because it wasn't settings.

The OSError is in fact true here and it's related to the container environment.  Somehow communications between the containers died and could not be recreated.

**Comment by bandwiches** [comment]
I found the cause of this and it's not exactly related to bad configuration.

When making changes to the LDAP(S) configuration via AWX GUI, the resulting change is not immediate.  The first 3-5 attempts after a change is made will result in this error.  Without making any changes, the error eventually resolves itself and logins began working.

Firewalls show the traffic reaching the LDAP(S) server every time, even when this error is present.  Configuration errors are generally present with different errors that aren't related to Transport (ie bad credentials).

TL;DR - when making changes to LDAP(S) give about 5 min before logging in, even then give it about 5 attempts to kick over for whatever reason.

Hope this helps someone in the future.
